enricher/epss/epss.go

```diff
@@ -0,0 +1,405 @@
+}
+
+// ParseEnrichment implements driver.EnrichmentUpdater.
+func (e *Enricher) ParseEnrichment(ctx context.Context, rc io.ReadCloser) ([]driver.EnrichmentRecord, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/epss/Enricher/ParseEnrichment")
+
+	defer func() {
+		_ = rc.Close()
+	}()
+
+	dec := json.NewDecoder(rc)
+	ret := make([]driver.EnrichmentRecord, 0, 250_000)
+	var err error
+
+	for {
+		var record driver.EnrichmentRecord
+		if err = dec.Decode(&record); err != nil {
+			break
+		}
+		ret = append(ret, record)
+	}
+
+	zlog.Debug(ctx).
+		Int("count", len(ret)).
+		Msg("decoded enrichments")
+
+	if !errors.Is(err, io.EOF) {
+		return nil, fmt.Errorf("error decoding enrichment records: %w", err)
+	}
+
+	return ret, nil
+}
+
+func (*Enricher) Name() string {
+	return epssName
+}
+
+func currentFeedURL() string {
+	currentDate := time.Now()
+	formattedDate := currentDate.Format("2006-01-02")
+	filePath := fmt.Sprintf("epss_scores-%s.csv.gz", formattedDate)
+
+	feedURL, err := url.Parse(DefaultFeed)
+	if err != nil {
+		panic(fmt.Errorf("invalid default feed URL: %w", err))
+	}
+
+	feedURL.Path = path.Join(feedURL.Path, filePath)
+	return feedURL.String()
+}
+
+func (e *Enricher) Enrich(ctx context.Context, g driver.EnrichmentGetter, r *claircore.VulnerabilityReport) (string, []json.RawMessage, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/epss/Enricher/Enrich")
+	m := make(map[string][]json.RawMessage)
+	erCache := make(map[string][]driver.EnrichmentRecord)
+
+	for id, v := range r.Vulnerabilities {
+		t := make(map[string]struct{})
+		ctx := zlog.ContextWithValues(ctx, "vuln", v.Name)
+
+		for _, elem := range []string{
+			v.Description,
+			v.Name,
+			v.Links,
+		} {
+			// Check if the element is non-empty before running the regex
+			if elem == "" {
+				zlog.Debug(ctx).Str("element", elem).Msg("skipping empty element")
+				continue
+			}
+
+			matches := enricher.CVERegexp.FindAllString(elem, -1)
+			if len(matches) == 0 {
+				zlog.Debug(ctx).Str("element", elem).Msg("no CVEs found in element")
+				continue
+			}
+			for _, m := range matches {
+				t[m] = struct{}{}
+			}
+		}
+
+		// Skip if no CVEs were found
+		if len(t) == 0 {
+			zlog.Debug(ctx).Msg("no CVEs found in vulnerability metadata")
+			continue
+		}
+
+		ts := make([]string, 0, len(t))
+		for m := range t {
+			ts = append(ts, m)
+		}
+		sort.Strings(ts)
+
+		cveKey := strings.Join(ts, "_")
+
+		rec, ok := erCache[cveKey]
+		if !ok {
+			var err error
+			rec, err = g.GetEnrichment(ctx, ts)
+			if err != nil {
+				return "", nil, err
+			}
+			erCache[cveKey] = rec
+		}
+
+		zlog.Debug(ctx).Int("count", len(rec)).Msg("found records")
+
+		// Skip if no enrichment records are found
+		if len(rec) == 0 {
+			zlog.Debug(ctx).Strs("cve", ts).Msg("no enrichment records found for CVEs")
+			continue
+		}
+
+		for _, r := range rec {
+			if _, exists := m[id]; !exists {
+				m[id] = []json.RawMessage{}
+			}
```

Comment:
I don't think this check is required. I think we can just do the `m[id] = append(m[id], r.Enrichment)` below (we can append something to a `nil` slice in Go)